Modules/mathmodule.c

```diff
@@ -2233,14 +2230,12 @@ loghelper(PyObject* arg, double (*func)(double))
         }
 
         x = PyLong_AsDouble(arg);
-        if (x == -1.0 && PyErr_Occurred()) {
-            if (!PyErr_ExceptionMatches(PyExc_OverflowError))
-                return NULL;
```

Comment:
@mdickinson, let me know if you think that this ``PyErr_ExceptionMatches()`` call should be kept like #112635.